src/index.js

```diff
@@ -121,27 +165,8 @@ class UglifyJsPlugin {
 							}
 
 							// Ensure that both conditions are functions
-							["preserve", "extract"].forEach(key => {
-								switch(typeof condition[key]) {
-									case "boolean":
-										var b = condition[key];
-										condition[key] = () => b;
-										break;
-									case "function":
-										break;
-									case "string":
-										if(condition[key] === "all") {
-											condition[key] = () => true;
-											break;
-										}
-										var regex = new RegExp(condition[key]);
-										condition[key] = (astNode, comment) => regex.test(comment.value);
-										break;
-									default:
-										regex = condition[key];
-										condition[key] = (astNode, comment) => regex.test(comment.value);
-								}
-							});
+							ensureConditionFunc(condition, "preserve");
```

Comment:
Please `-...-->Func<--` the function call should speak for itself :D, just `ensure(condition, ...)` maybe ?